base/applications/rapps/settings.cpp

```diff
@@ -201,29 +190,55 @@ FillDefaultSettings(PSETTINGS_INFO pSettingsInfo)
 
     PathAppendW(szDownloadDir.GetBuffer(MAX_PATH), L"\\RAPPS Downloads");
     szDownloadDir.ReleaseBuffer();
+}
 
-    CStringW::CopyChars(
-        pSettingsInfo->szDownloadDir, _countof(pSettingsInfo->szDownloadDir), szDownloadDir.GetString(),
-        szDownloadDir.GetLength() + 1);
+static VOID
+ValidateStringSettings(PSETTINGS_INFO pSettingsInfo)
+{
+    if (!pSettingsInfo->szDownloadDir[0])
+    {
+        CStringW szDownloadDir;
+        GetDefaultDownloadDirectory(szDownloadDir);
+
+        CStringW::CopyChars(pSettingsInfo->szDownloadDir, _countof(pSettingsInfo->szDownloadDir),
+                            szDownloadDir.GetString(), szDownloadDir.GetLength() + 1);
+    }
+}
+
+VOID
+FillDefaultSettings(PSETTINGS_INFO pSettingsInfo)
+{
+    ZeroMemory(pSettingsInfo, sizeof(SETTINGS_INFO));
```

Comment:
```suggestion
    ZeroMemory(pSettingsInfo, sizeof(*pSettingsInfo));
```

Reply:
I do know why you propose that (making the code auto-adapt to guessed future type-changes), but it's a bit harder to read I think.